### DISCUSSION TITLE
how to set many Workers?

### DISCUSSION THREAD

**Discussion by rustdreamer** [apalis-dev/apalis#20] [Q&A] [answered]
I want to build multiple async function works. How can I do this? The following cannot be achieved
```
Monitor::new()
    .register_with_count(6, move |_| {
        WorkerBuilder::new(worker_storage.clone())
            .layer(SentryJobLayer)
            .layer(TraceLayer::new())
            .build_fn(queue::send_email)
            .build_fn(queue::send_message)
            .build_fn(queue::and_more)
    })
```

**Comment by geofmureithi** [comment]
Do you mean something like this [example](https://github.com/geofmureithi/apalis/blob/master/examples/rest-api/src/main.rs#L320)?

**Reply by rustdreamer** [reply]
Similar to this, but many workers use Redis. I tried to write this, but I reported an error。
![image](https://user-images.githubusercontent.com/31054993/197720954-980fa057-e21c-4137-b498-72ce112ebbb6.png)

**Reply by geofmureithi** [reply]
Please share the complete code even if a gist. There is something you are missing. Also If you are using the same storage and layers then you might need to use an enum/steer.

**Reply by rustdreamer** [reply]
Not very familiar with enum/steer, do you have examples?

**Reply by geofmureithi** [reply]
in your case you can use an enum: 
```rust
enum Jobs {
    QueueFaceRecord(QueueFaceRecord),
    QueueTest(QueueTest)
}
```

**Comment by geofmureithi** [comment]
Related to #8

**Comment by geofmureithi** [comment, marked as answer]
FYI one worker can only take one job fn. If you need to use the same layers you could just use an enum.

**Comment by geofmureithi** [comment]
FYI you can use the steer feature by tower too:
https://docs.rs/tower/latest/tower/steer/index.html
